ui/imports/shared/popups/CommunityMembershipSetupDialog.qml

```diff
@@ -180,6 +137,52 @@ StatusStackModal {
             ]
         }
 
+        readonly property StatusFlatButton cancelRequestButton: StatusFlatButton {
+            text: qsTr("Cancel Membership Request")
+            type: StatusBaseButton.Type.Danger
+            onClicked: {
+                if (root.isInvitationPending) {
+                    root.cancelMembershipRequest()
+                    root.close()
+                    return
+                }
+            }
+        }
+
+        readonly property StatusButton shareAddressesButton: StatusButton {
+            text: {
+                if (root.accessType === Constants.communityChatOnRequestAccess) {
+                    if (d.selectedSharedAddressesCount === d.totalNumOfAddressesForSharing) {
+                        return qsTr("Share all addresses to join")
+                    }
+                    return qsTr("Share %n address(s) to join", "", d.selectedSharedAddressesCount)
+                }
+                return qsTr("Join %1").arg(root.communityName)
```

Comment:
This shouldn't be here, I believe this would conflict with another PR's of Sale's

Reply:
Yes, it will conflict but now there's no conflict on master that I can resolve and I will not modify it here to complicate the other PR!